### PR TITLE
[TT-6919] Feature/introspection exp check

### PR DESCRIPTION
**Pull request by furkansenharputlu**
This PR adds check for the cached introspection result expiration.

### CODE REVIEW

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4376/merge
Commit:  
Triggered by: pull_request (@furkansenharputlu)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/3360972015)

File: gateway/mw_external_oauth_test.go [hunk 1]
```diff
@@ -384,5 +385,15 @@ func TestExternalOAuthMiddleware_introspection(t *testing.T) {
 		_, _ = ts.Run(t, []test.TestCase{
 			{Path: "/get", Headers: headers, BodyMatch: "access token is not valid", Code: http.StatusUnauthorized},
 		}...)
+
+		t.Run("expired", func(t *testing.T) {
+			externalOAuthIntrospectionCache.DeleteAllKeys()
+			accessTokenActive = true
+			exp = time.Now().Add(-3 * time.Minute).Unix()
+			_, _ = ts.Run(t, []test.TestCase{
+				{Path: "/get", Headers: headers, BodyMatch: "/get", Code: http.StatusOK},
```
Comment (jeffy-mathew) [inline gateway/mw_external_oauth_test.go:394]: let's add a comment describing the test cases :)

**Review by jeffy-mathew — APPROVED** [review]

**Comment by Tyk-ITS** [comment]
**API tests result: failure** :no_entry_sign: 
Branch used: refs/pull/4376/merge
Commit: 29b3bc7ee00783cacddefd12dbab26451bb87a96 
Triggered by: pull_request (@furkansenharputlu)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/3361161322)

**Comment by Tyk-ITS** [comment]
**API tests result: failure** :no_entry_sign: 
Branch used: refs/pull/4376/merge
Commit: 331a4833486cf2f13569dff6273e818653f31218 
Triggered by: pull_request (@furkansenharputlu)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/3361172154)

File: gateway/mw_external_oauth.go [hunk 2]
```diff
@@ -241,6 +241,11 @@ func (k *ExternalOAuthMiddleware) introspection(accessToken string) (bool, strin
 		}
 	} else {
 		log.WithError(err).Debug("Found OAuth introspection result in the redis cache")
+
+		exp := claims["exp"].(float64)
```
Comment (titpetric) [inline gateway/mw_external_oauth.go:245]: Do we need to add ExpiresAtValidationSkew, like in JWT middleware?

Comment (furkansenharputlu) [reply gateway/mw_external_oauth.go:245]: It wasn’t in the contract and RFC so it is not implemented.

**Review by titpetric — APPROVED** [review]

**Comment by Tyk-ITS** [comment]
**API tests result: failure** :no_entry_sign: 
Branch used: refs/heads/master
Commit: 3914295981058137de66c22dbe8aac70b7dc1fc3 [TT-6919] Feature/introspection exp check (#4376)

* Check the expiration of cached introspection response

* Added a comment to explain the trick in the test cases
Triggered by: push (@furkansenharputlu)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/3361841648)

**Pull request merged**